tmt/steps/execute/internal.py

```diff
@@ -246,19 +214,75 @@ def _test_output_logger(
                 level=level,
                 topic=topic)
 
-        # Execute the test, save the output and return code
-        starttime = datetime.datetime.now(datetime.timezone.utc)
-        try:
-            output = guest.execute(
-                remote_command,
+        # This variable tracks execution time
+        # for test together with setup and cleanup scripts
+        time_spent = 0
+
+        def run_cmd(command: ShellScript, wrapper_filepath: Path) -> tmt.utils.CommandOutput:
+            nonlocal time_spent
+            friendly_command = str(command)
+            # Prepare the wrapper, push to guest
+            if test.framework == "shell":
+                command = ShellScript(f"{tmt.utils.SHELL_OPTIONS}; {command}")
+            self.debug('Test script', str(command), level=3)
+            self.write(wrapper_filepath, str(command), 'w')
+            wrapper_filepath.chmod(0o755)
+            guest.push(
+                source=wrapper_filepath,
+                destination=wrapper_filepath,
+                options=["-s", "-p", "--chmod=755"])
+
+            if self.get('interactive'):
+                command = ShellScript(
+                    TEST_WRAPPER_INTERACTIVE.format(remote_command=command))
+            else:
+                command = ShellScript(
+                    TEST_WRAPPER_NONINTERACTIVE.format(remote_command=command))
```

Comment:
Please, preserve `remote_command` here. Yes, `command` is the same type, and can be re-used, but by overwriting it the debugging gets harder. When listing locals, the original command would be no longer available.

Reply:
Fixed